[23기 한상웅] Refactor:오류 메세지 변경

tkddnd82 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 오류 메세지 변경
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
-중복되어지는 상황에서 처리해주는 오류메세지를 변경하였습니다.예) USERNAME_ALREADY_EXISTS
 username, email, phone_number, name 에서 변경 적용했습니다.

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [ x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [ x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

tkddnd82 committed:
feature/user5

S2-J1NG commented on users/views.py:
```diff
@@ -63,10 +63,10 @@ def post(self, request):
             password = data['password']
             
-            if not User.objects.filter(email=data['email']).exists():
+            if not User.objects.filter(username=data['username']).exists():
```
`email`에서 `username`으로 바뀐 이유 코멘트로 남겨주세요!

tkddnd82 replied:
기존에는 회원가입 할때 id를 email형식으로 했었는데 회원가입시에 email을 입력하는 칸이 있어서 굳이 그럴경우 중복되어지는 입력이 생기는거 같아서 프론트 분께서 username으로 id값을 변경 요청하셔서 수정하게 되었습니다.

S2-J1NG commented on users/views.py:
```diff
@@ -63,10 +63,10 @@ def post(self, request):
                 return JsonResponse({"MESSAGE" : "INVALID_USER"}, status=401)
                 
-            user = User.objects.get(email=data['email'])
+            user = User.objects.get(username=data['username'])
```
`username`으로 user 정보를 판별한다면, 같은 username을 쓰는 경우는 어떻게 처리하게 되나요??

tkddnd82 replied:
해당 부분 로직 추가해서 처리하도록 하겠습니다.

ghost commented:
이메일, 전화번호, 로그인 ID 와 같이 테이블에 고유 값 하나만 있어야 하는 경우 model에서 `unique` 옵션으로 설정할 수 있습니다! 
확인 후 `model.py`  적용여부는 팀원과 반드시 상의해주세요!

tkddnd82 commented:
> 이메일, 전화번호, 로그인 ID 와 같이 테이블에 고유 값 하나만 있어야 하는 경우 model에서 `unique` 옵션으로 설정할 수 있습니다!
> 확인 후 `model.py` 적용여부는 팀원과 반드시 상의해주세요!
>

tkddnd82 commented:
> 이메일, 전화번호, 로그인 ID 와 같이 테이블에 고유 값 하나만 있어야 하는 경우 model에서 `unique` 옵션으로 설정할 수 있습니다!
> 확인 후 `model.py` 적용여부는 팀원과 반드시 상의해주세요!

> -->넵 상의후에 적용여부 결정하도록 하겠습니다 : )

tkddnd82 commented:
> 이메일, 전화번호, 로그인 ID 와 같이 테이블에 고유 값 하나만 있어야 하는 경우 model에서 `unique` 옵션으로 설정할 수 있습니다!
> 확인 후 `model.py` 적용여부는 팀원과 반드시 상의해주세요!

> -->넵
 상의후에 적용여부 결정하도록 하겠습니다 : )

tkddnd82 committed:
Refactor:리뷰사항 반영
-기존 email형식의 아이디에서 username으로 변경(변경이유 코멘트에 남김)
-위의 변경 사항에 따라서 username에 유효성 검사 로직 추가
-멘토님께서 제안해주신 model에서 unique옵션 추가(팀원과 상의후에 추가하기로 하고 추가하였습니다.

tkddnd82 committed:
feature/user6

S2-J1NG commented on users/views.py:
```diff
@@ -31,6 +31,9 @@ def post(self, request):
                 return JsonResponse({'MESSAGE' : 'DUPLICATED EMAIL'}, status=400)
+
+            if User.objects.filter(username=data['username']).exists():
+                return JsonResponse({'MESSAGE' : 'DUPLICATED_USERNAME'}, status=400)
```
오류 메세지는 `USERNAME_ALREADY_EXISTS`가 더 적절해보입니다.
`username`이 이미 테이블에 있다면, 해당 `username`으로 등록한 사용자가 있으므로 다른 `username`을 사용하도록 프론트와 에러 처리 핸들링 논의해야 합니다.

tkddnd82 replied:
넵!

S2-J1NG commented on users/views.py:
```diff
@@ -31,6 +31,9 @@ def post(self, request):
+                return JsonResponse({'MESSAGE' : 'DUPLICATED_USERNAME'}, status=400)
                 
             if User.objects.filter(phone_number=data['phone_number']).exists():
                 return JsonResponse({'MESSAGE' : 'DUPLICATED PHONE_NUM'}, status=400)
```
여기도 적절한 오류 메세지 변경 해주세요!

tkddnd82 replied:
말씀해주신 username 오류메세지 부분 수정외에 혹시

> LGTM
> 포스트맨으로 에러 처리한 내용 잘 나오는 지 확인 하셨나요?? 에러 처리 한 부분도 의도대로 잘 나오는 지 확인 중요합니다!

네 오류메세지 변경한 부분 모두 처리 확인하였습니다.

ghost commented:
- `phonenumber` unique 처리 한 후 migrate 하셨나요?? `migrations` 파일에 해당 내용 없습니다!

- `password` 는 unique 설정을 왜 안 하는지 `bcrypt`와 `unique` 내용으로 블로그 주제로 적어도 좋을 거 같아요!
<a href = "https://docs.djangoproject.com/en/3.2/ref/models/fields/#unique"> unique </a> 옵션에 대한 내용은 `django` 공식 문서 참고해주세요!

tkddnd82 commented:
> * `phonenumber` unique 처리 한 후 migrate 하셨나요?? `migrations` 파일에 해당 내용 없습니다!
> * `password` 는 unique 설정을 왜 안 하는지 `bcrypt`와 `unique` 내용으로 블로그 주제로 적어도 좋을 거 같아요!
>   [ unique ](https://docs.djangoproject.com/en/3.2/ref/models/fields/#unique) 옵션에 대한 내용은 `django` 공식 문서 참고해주세요!

password에 unique처리한 후에는 migrations를 안한것 같습니다. 해당 사항 처리하도록 하겠습니다.
말씀해주신 내용도 이해한 후에 블로그 주제로 적도록 하겠습니다. 리뷰 감사합니다.

tkddnd82 committed:
Refactor:리뷰사항 수정 반영
-오류메시지 변경
-unique옵션 설정 부분 변경

ghost commented:
LGTM👏
포스트맨으로 에러 처리한 내용 잘 나오는 지 확인 하셨나요?? 에러 처리 한 부분도 의도대로 잘 나오는 지 확인 중요합니다!
